src/overlay/OverlayManagerImpl.cpp

```diff
@@ -1183,19 +1198,16 @@ OverlayManagerImpl::checkScheduledAndCache(
 }
 
 void
-OverlayManagerImpl::recvTransaction(StellarMessage const& msg,
+OverlayManagerImpl::recvTransaction(TransactionFrameBasePtr transaction,
                                     Peer::pointer peer, Hash const& index)
 {
     ZoneScoped;
-    auto transaction = TransactionFrameBase::makeTransactionFromWire(
-        mApp.getNetworkID(), msg.transaction());
+    releaseAssert(threadIsMain());
     if (transaction)
     {
         // record that this peer sent us this transaction
         // add it to the floodmap so that this peer gets credit for it
-        recvFloodedMsgID(msg, peer, index);
-
-        mTxDemandsManager.recordTxPullLatency(transaction->getFullHash(), peer);
+        recvFloodedMsgID(peer, index);
```

Comment:
I don't think we want to remove tx latency metrics in prod just yet, right? Maybe just hide this behind an if check based on if batching is enabled or not.

Reply:
good catch! fixed.